src/tests/test_continue.py

```diff
@@ -12,62 +12,97 @@
 from collections import OrderedDict
 from numpy import array
 
-class TestWaterTutorial(ForceBalanceTestCase):
-    def setUp(self):
-        super(ForceBalanceTestCase,self).setUp()
-        os.system('rm -rf test/files/test_continue.run')
-        os.makedirs('test/files/test_continue.run')
-        os.system('cp -r studies/001_water_tutorial/forcefield test/files/test_continue.run/forcefield')
-        os.system('cp -r studies/001_water_tutorial/targets test/files/test_continue.run/targets')
-        os.chdir('test/files')
-        os.system('cp -r test_continue.sav test_continue.in test_continue.tmp test_continue.run')
-        os.chdir('test_continue.run')
+# class TestWaterTutorial(ForceBalanceTestCase):
+class TestWaterTutirual:
+    # def setUp(self):
+    @classmethod
+    def setup_class(cls):
+        # super(ForceBalanceTestCase,self).setUp()
 
-    def tearDown(self):
-        os.chdir('..')
-        os.system('rm -rf test_continue.run')
-        super(ForceBalanceTestCase,self).tearDown()
+        # os.system('rm -rf tests/files/test_continue.run')
+        # os.makedirs('tests/files/test_continue.run')
+        # os.system('cp -r ../../studies/001_water_tutorial/forcefield tests/files/test_continue.run/forcefield')
+        # os.system('cp -r ../../studies/001_water_tutorial/targets tests/files/test_continue.run/targets')
+        # os.chdir('test/files')
+        # os.system('cp -r test_continue.sav test_continue.in test_continue.tmp test_continue.run')
+        # os.chdir('test_continue.run')
 
-    def runTest(self):
+        cls.cwd = os.path.dirname(os.path.realpath(__file__))
+        # copy folder 'files/test_liquid' into a new folder 'files/test_liquid.run'
+        os.chdir(os.path.join(cls.cwd, 'files'))
+        tmpfolder = os.path.join(cls.cwd, 'files', 'test_continue.run')
+        if os.path.isdir(tmpfolder):
+            shutil.rmtree(tmpfolder)
+        os.mkdir(tmpfolder)
+        source_folder = os.path.join(cls.cwd, '../../studies/001_water_tutorial')
+        shutil.copytree(os.path.join(source_folder, 'forcefield'), os.path.join(tmpfolder, 'forcefield'))
+        shutil.copytree(os.path.join(source_folder, 'targets'), os.path.join(tmpfolder, 'targets'))
+        # os.system('cp -r test_continue.sav test_continue.in test_continue.tmp test_continue.run')
+        shutil.copy("test_continue.sav", tmpfolder)
+        shutil.copy("test_continue.in", tmpfolder)
+        shutil.copytree("test_continue.tmp", os.path.join(tmpfolder,"test_continue.tmp"))
+        os.chdir(tmpfolder)
+
+    # def tearDown(self):
+    @classmethod
+    def teardown_class(cls):
+        # os.chdir('..')
```

Comment:
Is it OK that we don't change directories during teardown? Do the tests end up in the wrong place if we don't return to the same directory after each one?

Reply:
We should change the `setup` function that runs before tests to remember which directory it starts in (`self.start_directory = os.getcwd()`) and then return to that directory during teardown (`os.chdir(self.start_directory)`). Otherwise it will be easy to get lost in directories if we always _assume_ that we go down two directories before a test, and then back up two directories after it. By defining an absolute path, we avoid potentially making that mistake.

Reply:
Yes, I agree with you.  The FB process does change folders as it runs, and that was a design choice I made early on (for better or worse) because it has to create and manage a complex directory structure.  We should have a unified condition for all tests that they must finish in the same folder that they're started in.